dbms/src/Parsers/ParserInsertQuery.cpp

```diff
@@ -155,6 +182,7 @@ bool ParserInsertQuery::parseImpl(Pos & pos, ASTPtr & node, Expected & expected)
 
     query->columns = columns;
     query->select = select;
+    query->input_function = input_function;
```

Comment:
Better to move it from AST. Because it isn't an AST detail.
And analyze the existence of input function later.